crates/ty_python_semantic/resources/mdtest/type_properties/constraints.md

```diff
@@ -626,11 +628,50 @@ def f[T, U]():
     reveal_type((t1 | t2) & (u1 | u2))
 ```
 
+We might simplify a BDD so much that we can no longer see the constraints that we used to construct
+it!
+
+```py
+from typing import Never
+from ty_extensions import constraint_set_domain, static_assert
+
+def f[T]():
+    t_int = range_constraint(Never, T, int)
+    t_bool = range_constraint(Never, T, bool)
+
+    # `T ≤ bool` implies `T ≤ int`: if a type satisfies the former, it must always satisfy the
+    # latter. We can turn that into a constraint set, using the equivalence `p → q == ¬p ∨ q`:
+    implication = ~t_bool | t_int
+    # revealed: ty_extensions.ConstraintSet[always]
+    reveal_type(implication)
+    static_assert(implication)
+
+    # However, because of that implication, some inputs aren't valid: it's not possible for
+    # `T ≤ bool` to be true and `T ≤ int` to be false. This is reflected in the constraint set's
+    # "domain", which maps valid inputs to `true` and invalid inputs to `false`.
+    # TODO: Under the covers, the domain BDD actually does include the `¬(T@f ≤ bool)` constraint,
+    # but it's currently being hidden by our display simplification logic.
+    # TODO: revealed: ty_extensions.ConstraintSet[((T@f ≤ int) ∧ ¬(T@f ≤ bool))]
+    # revealed: ty_extensions.ConstraintSet[(T@f ≤ int)]
+    reveal_type(constraint_set_domain(implication))
```

Comment:
Ugh... I wanted to display this without simplifying the DNF so that we could see (and test!) that its internal structure is not just the `AlwaysTrue` terminal. But the display simplification rules are important to get consistent output even when BDD variable ordering changes. So this is a flaky assertion, as can be seen in the CI failures.

For now, I think I'm going to back out the parts that let us display BDD domains in the mdtests. The equality test below still lets us test that the domain is not just `AlwaysTrue`, even if we can't display it nicely.